### PR TITLE
Abstract states into platform/static-data

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issue:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/7128

This PR removes the `country` dropdown and abstracts `STATES.json` into `platform/static-data` so that we can reuse it between all apps if needed/desired.

## Testing done
Updated unit tests.

## Screenshots
image

## Acceptance criteria
- [x] Remove `country` dropdown and field support.
- [x] Abstract `STATES.json` into `platform/static-data`.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by kelsonic** [comment]
@bradl3yC @erikphansen Hey! So in this PR we've created an analogous `STATES.json` in `platform/static-data` in case you want to use that instead of https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/user/profile/vet360/constants/states.json 🙂Not necessary to do but just wanted to give you both the heads up that it's an option now!

**Review by ncksllvn — APPROVED** [review]
LGTM! Great idea with creating the states list in `platform/`

File: src/applications/yellow-ribbon/containers/SearchForm/index.jsx [hunk 1]
```diff
@@ -3,11 +3,10 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import URLSearchParams from 'url-search-params';
-import classNames from 'classnames';
 import map from 'lodash/map';
 // Relative imports.
 import ErrorableCheckbox from '@department-of-veterans-affairs/formation-react/ErrorableCheckbox';
-import STATES from '../../constants/STATES.json';
+import STATES from '../../../../platform/static-data/STATES.json';
```
[the remark this answers is not in the document]
Comment (kelsonic) [reply src/applications/yellow-ribbon/containers/SearchForm/index.jsx]: Updated! 🙂

**Comment by kelsonic** [comment]
@department-of-veterans-affairs/frontend-review-group Ready for review! 🎉

**Review by rianfowler — APPROVED** [review]

**Pull request merged**